enhancements/authentication/oauth-resource-storage.md

```diff
@@ -0,0 +1,253 @@
+    // +sensitive
+    RefreshToken string `json:"refreshToken,omitempty" protobuf:"bytes,9,opt,name=refreshToken"`
+
+    // InactivityTimeoutSeconds is the value in seconds, from the
+    // CreationTimestamp, after which this token can no longer be used.
+    // The value is automatically incremented when the token is used.
+    InactivityTimeoutSeconds int32 `json:"inactivityTimeoutSeconds,omitempty" protobuf:"varint,10,opt,name=inactivityTimeoutSeconds"`
+
+    // SecretToken is the second half of the secret information a user has to pass as bearer token
+    // in order to authorize with the API server. It is concatenated with the name of the 
+    // OAuthAccessToken object.
+    // For old tokens, this field might be empty.
+    // +sensitive
+    // +optional
+    SecretToken string `json:"secretToken,omitempty" protobuf:"bytes,11,opt,name=secretToken"`
+}
+
+// OAuthAuthorizeToken describes an OAuth authorization token
+type OAuthAuthorizeToken struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+    // ClientName references the client that created this token.
+    ClientName string `json:"clientName,omitempty" protobuf:"bytes,2,opt,name=clientName"`
+
+    // ExpiresIn is the seconds from CreationTime before this token expires.
+    ExpiresIn int64 `json:"expiresIn,omitempty" protobuf:"varint,3,opt,name=expiresIn"`
+
+    // Scopes is an array of the requested scopes.
+    Scopes []string `json:"scopes,omitempty" protobuf:"bytes,4,rep,name=scopes"`
+
+    // RedirectURI is the redirection associated with the token.
+    RedirectURI string `json:"redirectURI,omitempty" protobuf:"bytes,5,opt,name=redirectURI"`
+
+    // State data from request
+    State string `json:"state,omitempty" protobuf:"bytes,6,opt,name=state"`
+
+    // UserName is the user name associated with this token
+    UserName string `json:"userName,omitempty" protobuf:"bytes,7,opt,name=userName"`
+
+    // UserUID is the unique UID associated with this token. UserUID and UserName must both match
+    // for this token to be valid.
+    UserUID string `json:"userUID,omitempty" protobuf:"bytes,8,opt,name=userUID"`
+
+    // CodeChallenge is the optional code_challenge associated with this authorization code, as described in rfc7636
+    CodeChallenge string `json:"codeChallenge,omitempty" protobuf:"bytes,9,opt,name=codeChallenge"`
+
+    // CodeChallengeMethod is the optional code_challenge_method associated with this authorization code, as described in rfc7636
+    CodeChallengeMethod string `json:"codeChallengeMethod,omitempty" protobuf:"bytes,10,opt,name=codeChallengeMethod"`
+
+    // SecretToken is the second half of the secret information a user has to pass as bearer token
+    // in order to authorize with the API server. It is concatenated with the name of the 
+    // OAuthAuthorizeToken object.
+    // For old tokens, this field might be empty.
+    // +sensitive
```

Comment:
is this tag interpreted by some components?

Reply:
no